src/main/kotlin/car/business/Cars.kt

```diff
@@ -1,33 +1,31 @@
 package car.business
 
-import kotlin.random.Random
-import kotlin.streams.toList
+import car.energy.EnergyProvider
+import car.energy.RandomEnergyProvider
 
-class Cars(amount: Int, random: Random = Random) {
-    private val RANDOM_MAX_SIZE = 10
+class Cars(names: List<String>, energyProvider: EnergyProvider = RandomEnergyProvider()) {
```

Comment:
아래와 같은 주 생성자(primary constructor)를 만든 후에 자기 생성자 호출하는 `this()`를 사용하면 어떨까요?

```suggestion
class Cars(private val cars: List<Car>, private val energyProvider: EnergyProvider = RandomEnergyProvider()) {
```